website/docs/reference/resource-configs/contract.md

```diff
@@ -18,9 +18,13 @@ This functionality is new in v1.5.
 
 When the `contract` configuration is enforced, dbt will ensure that your model's returned dataset exactly matches the attributes you have defined in yaml:
 - `name` and `data_type` for every column
-- additional [`constraints`](resource-properties/constraints), as supported for this materialization + data platform
+- Additional [`constraints`](resource-properties/constraints), as supported for this materialization + data platform
 
-The `data_type` defined in your yaml file must match a data type your data platform recognizes. dbt does not do any type aliasing itself; if your data platform recognizes both `int` and `integer` as corresponding to the same type, then they will return a match.
+This is to ensure that downstream queriers of your model—inside and outside dbt—have a predictable and consistent set of columns to use in their analyses. Even a subtle change in data type, such as from `boolean` (`true`/`false`) to `integer` (`0`/`1`), could cause queries to fail in surprising ways.
```

Comment:
I'm going to switch `downstream queriers` &rarr; `people querying your model downstream`. I do think it's important that we're talking about people here, and not just impersonal queries/systems.

> both within and outside dbt

Agree with "both." I'm going to keep this parenthetical inside m-dashes for emphasis, rather than just an aside, because I think it's a really important point. Doug seems to agree :) https://github.com/dbt-labs/docs.getdbt.com/pull/3249#discussion_r1175370722